packages/react-vapor/src/components/color-picker/ColorPicker.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {ChromePicker, ChromePickerProps, ColorResult} from 'react-color';
+import {connect} from 'react-redux';
+import {isString, noop, uniqueId} from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {addInput, changeInputValue, removeInput} from '../input/InputActions';
+import {InputConnected} from '../input/InputConnected';
+import {InputSelectors} from '../input/InputSelectors';
+
+export interface IColorPickerProps extends ChromePickerProps {
+    id?: string;
+    defaultColor?: string;
+}
+
+export interface IColorPickerDispatchProps {
+    onDestroy?: () => void;
+    componentDidMount?: (value?: string, valid?: boolean, disabled?: boolean) => void;
+    onChangeComplete?: (colorPicked: ColorResult) => void;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IColorPickerProps) => {
+    const color = InputSelectors.getValue(state, {id: ownProps.id}) || ownProps.defaultColor;
+    return {
+        color,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IColorPickerProps): IColorPickerDispatchProps => ({
+    componentDidMount: (value: string = '', valid = true, disabled = false) => {
+        dispatch(addInput(ownProps.id, value, valid, disabled));
+        dispatch(changeInputValue(ownProps.id, value, true));
+    },
+    onDestroy: () => dispatch(removeInput(ownProps.id)),
+    onChangeComplete: (colorPicked: ColorResult) => dispatch(changeInputValue(ownProps.id, colorPicked.hex, true)),
+});
+
+export class ColorPicker extends React.Component<IColorPickerProps & IColorPickerDispatchProps> {
```

Comment:
We stopped creating both version (connected & not connected) of components since we only used the connected ones in the admin. The non-connected version should not be exported

```typescript
class ColorPickerDisconnected extends React.PureComponent<...> {
...
export const ColorPicker = connect(
```

(also your component uses the InputConnected so it wouldn't work in a non-connected environment)